packages/tpetra/core/test/CrsGraph/CrsGraph_UnitTests_Swap.cpp

```diff
@@ -44,348 +44,434 @@
 #include <set>
 #include <unordered_map>
 #include <utility>
+#include <tuple>
+#include <vector>
 
 #include <Tpetra_ConfigDefs.hpp>
 #include <Tpetra_CrsGraph.hpp>
+#include <Tpetra_CrsMatrix.hpp>
 #include <Tpetra_Details_Behavior.hpp>
 #include <Tpetra_TestingUtilities.hpp>
 #include <type_traits>      // std::is_same
 
-namespace {      // (anonymous)
 
-using std::endl;
-using std::max;
-using std::min;
-using Teuchos::arcp;
-using Teuchos::arcpClone;
-using Teuchos::Array;
-using Teuchos::ArrayRCP;
-using Teuchos::ArrayView;
-using Teuchos::as;
-using Teuchos::Comm;
-using Teuchos::null;
-using Teuchos::outArg;
-using Teuchos::ParameterList;
-using Teuchos::parameterList;
-using Teuchos::RCP;
-using Teuchos::rcp;
-using Teuchos::tuple;
-using Teuchos::VERB_EXTREME;
-using Teuchos::VERB_HIGH;
-using Teuchos::VERB_LOW;
-using Teuchos::VERB_MEDIUM;
-using Teuchos::VERB_NONE;
-using Tpetra::DynamicProfile;
-using Tpetra::ProfileType;
-using Tpetra::StaticProfile;
-using Tpetra::TestingUtilities::getDefaultComm;
-typedef Tpetra::global_size_t GST;
+namespace Tpetra {
 
-double      errorTolSlack = 1e+1;
-std::string filedir;
+template<class LocalOrdinal, class GlobalOrdinal, class Node>
+class crsGraph_Swap_Tester
+{
+    using Scalar                   = int;                                                   // Not used here, kept for consistency.
```

Comment:
It is used, but just for the structs I'm passing into the graph generator in the test so warnings won't get generated. It's kept here mostly so that I can just use the same thing for generating my graphs in both this test and the CrsMatrix swap test.  
In the CrsMatrix Swap test, the scalar parameter is coming in from Trilinos so I don't have this line in that test. For here, it's ok because I'm just using it to keep the struct happy.